src/pyclass.rs

```diff
@@ -44,8 +42,24 @@ pub(crate) unsafe fn default_new<T: PyTypeInfo>(
             unreachable!("Subclassing native types isn't support in limited API mode");
         }
     }
+
     let alloc = get_type_alloc(subtype).unwrap_or(ffi::PyType_GenericAlloc);
-    alloc(subtype, 0) as _
+    let ptr = alloc(subtype, 0);
+
+    // Workaround for Python issue 35810; no longer necessary in Python 3.8
+    #[cfg(not(Py_3_8))]
+    {
+        use crate::once_cell::GILOnceCell;
+
+        // Must check version at runtime because of abi3 wheels which could run against a higher
+        // version than the config.
+        static IS_NOT_PYTHON_3_8: GILOnceCell<bool> = GILOnceCell::new();
```

Comment:
How about checking the current refcount instead of the version?

Reply:
I couldn't find a way to detect this case using the type reference count - it increases by one on all versions I tested.

I guess I need to make this check only with `Py_Limited_API`, so I could avoid checking the version in more cases.